sql/core/src/test/scala/org/apache/spark/sql/DatasetSuite.scala

```diff
@@ -2841,6 +2860,265 @@ class DatasetSuite extends QueryTest
     checkDataset(Seq(seqMutableSet).toDS(), seqMutableSet)
     checkDataset(Seq(mapMutableSet).toDS(), mapMutableSet)
   }
+
+  // below tests are related to SPARK-49960 and TransformingEncoder usage
+  test("Incorrect derived nullability with TransformingEncoder - non nullable") {
+    val sparkI = spark
+    type T = Tuple2[Seq[Seq[Int]], Seq[Int]]
+    val data: Seq[T] = Seq( ( Seq( Seq(1, 2, 3) ), Seq(1, 2, 3) ) )
+    // for reference only
+    val sparkDataTypeOG = {
+      import sparkI.implicits._
```

Comment:
NIT: Don't we import `testImplicits._` here?

Reply:
I've removed all of the "for reference only" sections, they were only for debug and I should have removed the beforehand